mewarpx/mewarpx/assemblies.py

```diff
@@ -395,23 +430,24 @@ def _get_transformation_matrix(self):
         return transformation_matrix
 
     def _transform_coordinates(self, points_in):
-
+        """Transforms the coordinates of the particles using the
+        transformation matrix of the rectangle"""
         points = np.ones((len(points_in), 3))
-        points[:,0] = points_in[:,0]
-        points[:,1] = points_in[:,1]
+        points[:, 0] = points_in[:, 0]
+        points[:, 1] = points_in[:, 1]
 
         points = points.dot(self.transformation_matrix.T)
 
-        return points[:,0:2]
+        return points[:, 0:2]
 
     def _sort_particles(self, points):
-
+        """Finds the regions of the rectangle that each point belongs to"""
         # the checks below isolate region in which the given position cannot
-        # be, for example is c01 is True for a specific position it means
+        # be, for example if c01 is True for a specific position it means
         # that position cannot be in region 0 or 1.
-        c01 = np.array(points[:,1] > -abs(points[:,0]), dtype=int)
-        c45 = np.array(points[:,1] - self.scaled_h + 2.0 < abs(points[:,0]), dtype=int)
-        c23 = np.array(points[:,0] <= 0, dtype=int)
+        c01 = np.array(points[:, 1] > -abs(points[:, 0]), dtype=int)
+        c45 = np.array(points[:, 1] - self.scaled_h + 2.0 < abs(points[:, 0]), dtype=int)
+        c23 = np.array(points[:, 0] <= 0, dtype=int)
```

Comment:
My last point of confusion - We don't have a region 4 or 5, so what is c45 doing? I'm having trouble mapping any of these onto the four regions nicely.

Reply:
I think @roelof-groenewald can probably shed some more light on this. From my understanding, there used to be 8 regions, but we realized we could combine some regions to reduce the amount of calculations, and this naming scheme is left over from that. So 01 is just region 0, 23 is region 1, 45 is region 2, and 67 is region 3. I will rename these variables to reflect that change.

Reply:
That is correct. Initially the algorithm picked out the regions inside the rectangle as well and would give a normal to the inside of the rectangle but since we only use normals facing outwards we were able to reduce the number of regions to only 4 by combining regions inside the rectangle with appropriate ones outside.